components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -107,7 +110,8 @@ export class PrebuildManager {
 
             span.setTag("starting", true);
             await this.workspaceStarter.startWorkspace({ span }, workspace, user, [], {excludeFeatureFlags: ['full_workspace_backup']});
-            return { wsid: workspace.id, done: false };
+            const prebuild = (await prebuildPromise)!;
```

Comment:
I'm just slightly worried about this code:
- Logically, `findPrebuildByWorkspaceID(workspace.id)` should always find a `prebuild`, given that the workspace was created from a `StartPrebuildContext` context
- However, what if there is an edge case, or a new bug in the future 😱

Reply:
indeed, it seems to lack a contract here, i.e. it seems easy to break because it relies on assumptions, though it's not changed frequently.

perhaps a dedicated `workspaceFactory.createPrebuild(..., context): { ws: Workspace, pb: Prebuild }` would document the requirements more maintainable? just thinking out loud...